pkg/imageinspector/secretfetcher.go

```diff
@@ -2,49 +2,83 @@ package imageinspector
 
 import (
 	"context"
-	"sync"
+	"time"
 
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 
+	"github.com/kubeshop/testkube/pkg/cache"
+	"github.com/kubeshop/testkube/pkg/log"
+
 	"github.com/kubeshop/testkube/pkg/secret"
 )
 
+type SecretFetcherOption func(*secretFetcher)
+
+// WithSecretCacheTTL sets the time to live for the cached secrets.
+func WithSecretCacheTTL(ttl time.Duration) SecretFetcherOption {
+	return func(s *secretFetcher) {
+		s.ttl = ttl
+	}
+}
+
 type secretFetcher struct {
 	client secret.Interface
-	cache  map[string]*corev1.Secret
-	mu     sync.RWMutex
+	cache  cache.Cache[*corev1.Secret]
+	ttl    time.Duration
 }
 
-func NewSecretFetcher(client secret.Interface) SecretFetcher {
-	return &secretFetcher{
+func NewSecretFetcher(client secret.Interface, cache cache.Cache[*corev1.Secret], opts ...SecretFetcherOption) SecretFetcher {
+	s := &secretFetcher{
 		client: client,
-		cache:  make(map[string]*corev1.Secret),
+		cache:  cache,
+	}
+	for _, opt := range opts {
+		opt(s)
 	}
+	return s
 }
 
 func (s *secretFetcher) Get(ctx context.Context, name string) (*corev1.Secret, error) {
-	// Get cached secret
-	s.mu.RLock()
-	if v, ok := s.cache[name]; ok {
-		s.mu.RUnlock()
-		return v, nil
+	if s.ttl > 0 {
```

Comment:
Looks like a clash with TTL from cache - zero here won't cache

Reply:
I thought about adding a new ennvvar, but it seemed simpler to just use 0 for no cache, and you could set the cache duration to a really high duration for "infinite".

I don't have a strong opinion on this.

Reply:
In this comment, I'm only showing that there is clash between `InMemoryCache` and `SecretFetcher` when `ttl: 0`:
* `SecretFetcher` - when `ttl: 0` it won't use cache (as `(0 > 0) == false`)
* `InMemoryCache` - when `ttl: 0` it will cache indefinitely

Reply:
oh, thanks, good catch